src/os/ui/draw/basedrawcontrols.js

```diff
@@ -290,15 +294,18 @@ os.ui.draw.BaseDrawControlsCtrl.prototype.initControlMenu = function() {
     mi.removeChild('Whole World');
     mi.removeChild('drawMenuSeparator');
   }
-  if (this.isCountryEnabled() && !this['hideExtraControls']) {
-    mi.addChild({
-      label: 'Country Border',
-      eventType: os.ui.menu.draw.EventType.COUNTRY_BORDER,
-      tooltip: 'Add a country border as an area',
-      icons: ['<i class="fa fa-fw fa-globe"></i> '],
-      handler: os.ui.menu.draw.handleDrawEvent,
-      sort: 130
-    });
+
+  if (!this['hideExtraControls']) {
+    var sort = 130;
+    for (const [, service, settings] of os.ogc.registry.getInstance().entries()) {
+      // must be configured in settings.json and have MenuItemOptions (i.e. settings)
+      if (service.isConfigured() && settings) {
```

Comment:
Here and everywhere else that a menu is being registered from the registry should be refactored a bit.

The way I envision it, these menu items should be thought of as being injected by a plugin that represents the potential providers of external areas, and therefore references to country borders or bodies of water should never appear here or anywhere else besides the classes that register them to the registry.

Part of what they register should be their menu handler functions, so that those can simply be set as the `handler` property on the menu items here.